pkg/porter/credentials.go

```diff
@@ -211,6 +207,43 @@ func (p *Porter) ShowCredential(opts CredentialShowOptions) error {
 // GetCredentialSourceValueAndType takes a given credentials.Source struct and
 // returns the source value itself as well as source type, e.g., 'path', 'env', etc.,
 // both in their string forms
-func GetCredentialSourceValueAndType(cs credentials.Source) (string, string) {
+func GetCredentialSourceValueAndType(cs credentials.Source) (value string, key string) {
 	return cs.Value, cs.Key
 }
+
+// CredentialDeleteOptions represent options for Porter's credential delete command
+type CredentialDeleteOptions struct {
+	Name string
+}
+
+// DeleteCredential deletes the credential set corresponding to the provided
+// names.
+func (p *Porter) DeleteCredential(opts CredentialDeleteOptions) error {
+	err := p.Credentials.Delete(opts.Name)
+	if err == crud.ErrRecordDoesNotExist {
+		fmt.Fprintln(p.Out, "credential set does not exist")
```

Comment:
Currently when this situation happens (Delete something that's not there) we have been only [logging](https://github.com/deislabs/porter/blob/master/CONTRIBUTING.md#logging) this message when p.Debug is set, and then printing to p.Err.

If you have feels about that behavior, it's up for debate, but if you don't, let's change this here to be consistent with the current behavior.